master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -116,6 +116,8 @@ func (a ResourceManager) ResolveResourcePool(
 			resp, err := a.GetDefaultAuxResourcePool(actorCtx, req)
 			if err != nil {
 				return "", fmt.Errorf("defaulting to aux pool: %w", err)
+			} else if resp.PoolName == "" {
+				return "", fmt.Errorf("no resource pool specified and no default pool set")
```

Comment:
Can we have `GetDefaultAuxResourcePool` return an 404 (error) instead of using `resp.PoolName == ""` as an in-band error? IMO, if you call "GetDefaultAuxResourcePool" and you _don't_ get a default back, that should be an error.

Reasoning is, if we do this, `go vet` will make sure people check that error, but if we don't, unless you read the signature carefully you may not know `resp.PoolName` can be empty.

Also see https://github.com/golang/go/wiki/CodeReviewComments#in-band-errors